linuxPackages.xone: 0.3-unstable-2024-04-25 -> 0.3-unstable-2024-12-23; switch to dlundqvist fork, drop patches for newer kernels, add 'fazzi' as maintainer

fxzzi opened the pull request:
Currently, we are using upstream https://github.com/medusalix/xone as the source for our xone nixpkg. This has caused a few issues, like the fact that we have to pull in some patches ourselves to make the module successfully build for newer kernels (6.11+).

We should follow what the official xone support server is suggesting, which is to use dlundqvist's fork of the project. 
![image](https://github.com/user-attachments/assets/cabfbed6-575d-4815-bf35-895ac8b6704b)

This allows us to remove the patches we were originally applying (they are upstream in this fork), and also puts us on a more up to date path for the package. It is clear that the original repo is no longer being worked on, considering it hasn't had an update in 10 months, the creator's (medusalix) last message on the support server was on 2024/05/13, and the AUR package maintainer (which was previously me when I was running arch), has now moved the base package over to this fork as well. https://aur.archlinux.org/packages/xone-dkms-git

The comment from the new maintainer states: 
![image](https://github.com/user-attachments/assets/19a9906b-b96f-4a48-a0a9-b2b6d04540c1)

This fork also contains a few minor changes including additional support for some devices, and some other minor fixes.

Finally, I have also added myself as a maintainer of this package, just in case the driver ever breaks again for a newer kernel release.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

9999years reviewed (approved):
LGTM, thanks for this!

JohnRTitor commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 384828`

Logs: https://github.com/JohnRTitor/nixpkgs-review-gha/actions/runs/13920075762


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.xone</li>
    <li>linuxKernel.packages.linux_5_10_hardened.xone</li>
    <li>linuxKernel.packages.linux_5_4.xone</li>
    <li>linuxKernel.packages.linux_5_4_hardened.xone</li>
    <li>linuxPackages_5_10_hardened.xone</li>
    <li>linuxPackages_5_4_hardened.xone</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 17 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.xone</li>
    <li>linuxPackages_5_15_hardened.xone (linuxKernel.packages.linux_5_15_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_1.xone</li>
    <li>linuxKernel.packages.linux_6_11.xone</li>
    <li>linuxPackages_6_11_hardened.xone (linuxKernel.packages.linux_6_11_hardened.xone)</li>
    <li>linuxPackages.xone (linuxKernel.packages.linux_6_12.xone)</li>
    <li>linuxPackages_hardened.xone (linuxPackages_6_12_hardened.xone)</li>
    <li>linuxPackages_latest.xone (linuxKernel.packages.linux_6_13.xone)</li>
    <li>linuxPackages_6_1_hardened.xone (linuxKernel.packages.linux_6_1_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_6.xone</li>
    <li>linuxPackages_6_6_hardened.xone (linuxKernel.packages.linux_6_6_hardened.xone)</li>
    <li>linuxPackages_latest-libre.xone (linuxKernel.packages.linux_latest_libre.xone)</li>
    <li>linuxPackages-libre.xone (linuxKernel.packages.linux_libre.xone)</li>
    <li>linuxPackages_lqx.xone (linuxKernel.packages.linux_lqx.xone)</li>
    <li>linuxPackages_xanmod.xone (linuxKernel.packages.linux_xanmod.xone)</li>
    <li>linuxPackages_xanmod_latest.xone (linuxKernel.packages.linux_xanmod_latest.xone, linuxPackages_xanmod_stable.xone)</li>
    <li>linuxPackages_zen.xone (linuxKernel.packages.linux_zen.xone)</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 16 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.xone</li>
    <li>linuxKernel.packages.linux_5_10_hardened.xone</li>
    <li>linuxKernel.packages.linux_5_4.xone</li>
    <li>linuxKernel.packages.linux_5_4_hardened.xone</li>
    <li>linuxKernel.packages.linux_lqx.xone</li>
    <li>linuxKernel.packages.linux_xanmod.xone</li>
    <li>linuxKernel.packages.linux_xanmod_latest.xone</li>
    <li>linuxKernel.packages.linux_xanmod_stable.xone</li>
    <li>linuxKernel.packages.linux_zen.xone</li>
    <li>linuxPackages_5_10_hardened.xone</li>
    <li>linuxPackages_5_4_hardened.xone</li>
    <li>linuxPackages_lqx.xone</li>
    <li>linuxPackages_xanmod.xone</li>
    <li>linuxPackages_xanmod_latest.xone</li>
    <li>linuxPackages_xanmod_stable.xone</li>
    <li>linuxPackages_zen.xone</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>linuxPackages_hardened.xone (linuxPackages_6_12_hardened.xone)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.xone</li>
    <li>linuxPackages_5_15_hardened.xone (linuxKernel.packages.linux_5_15_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_1.xone</li>
    <li>linuxKernel.packages.linux_6_11.xone</li>
    <li>linuxPackages_6_11_hardened.xone (linuxKernel.packages.linux_6_11_hardened.xone)</li>
    <li>linuxPackages.xone (linuxKernel.packages.linux_6_12.xone)</li>
    <li>linuxPackages_latest.xone (linuxKernel.packages.linux_6_13.xone)</li>
    <li>linuxPackages_6_1_hardened.xone (linuxKernel.packages.linux_6_1_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_6.xone</li>
    <li>linuxPackages_6_6_hardened.xone (linuxKernel.packages.linux_6_6_hardened.xone)</li>
    <li>linuxPackages_latest-libre.xone (linuxKernel.packages.linux_latest_libre.xone)</li>
    <li>linuxPackages-libre.xone (linuxKernel.packages.linux_libre.xone)</li>
  </ul>
</details>

Cryolitia reviewed (approved):
LGTM

JohnRTitor reviewed (commented):
Please separate each logical changes to individual commits.

fxzzi committed:
linuxPackages.xone: add 'fazzi' as maintainer

fxzzi commented:
> Please separate each logical changes to individual commits.

Should be resolved by separating the commit where I add myself as a maintainer, format the package, and then the rest. Let me know if this works for you.

fxzzi committed:
linuxPackages.xone: 0.3-unstable-2024-04-25 -> 0.3-unstable-2024-12-23

switch to a fork of the original repo from dlundqvist. It already
contains commits to fix builds for 6.11+ kernels, and contains other
minor changes. This allows us to drop the patches which we were
previously applying. The upstream repo from medusalix seems to be abandoned,
and the maintainer of said repo hasn't had any communication on the
support discord server since March 2024.

JohnRTitor reviewed (commented):
Nixfmt RFC style commit is very minimal (just one line) change? So you can just merge it with the last one. 

Also change that commit's message to `linuxPackages.xone: 0.3-unstable-2024-04-25 -> 0.3-unstable-2024-12-23` and explain why are you switching to the fork.

fxzzi commented:
> Nixfmt RFC style commit is very minimal (just one line) change? So you can just merge it with the last one.

yeah I have no idea what even happened there and why the line was removed then added again right after. I've corrected it and squashed down to two commits.

> Also change that commit's message to `linuxPackages.xone: 0.3-unstable-2024-04-25 -> 0.3-unstable-2024-12-23` and explain why are you switching to the fork.

sure. lets hope i added the comment to the commit properly haha

JohnRTitor commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 384828`

Logs: https://github.com/JohnRTitor/nixpkgs-review-gha/actions/runs/14064254305


---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.xone</li>
    <li>linuxKernel.packages.linux_5_10_hardened.xone</li>
    <li>linuxKernel.packages.linux_5_4.xone</li>
    <li>linuxKernel.packages.linux_5_4_hardened.xone</li>
    <li>linuxPackages_5_10_hardened.xone</li>
    <li>linuxPackages_5_4_hardened.xone</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 18 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.xone</li>
    <li>linuxPackages_5_15_hardened.xone (linuxKernel.packages.linux_5_15_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_1.xone</li>
    <li>linuxPackages.xone (linuxKernel.packages.linux_6_12.xone)</li>
    <li>linuxPackages_hardened.xone (linuxPackages_6_12_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_13.xone</li>
    <li>linuxPackages_6_13_hardened.xone (linuxKernel.packages.linux_6_13_hardened.xone)</li>
    <li>linuxPackages_latest.xone (linuxKernel.packages.linux_6_14.xone)</li>
    <li>linuxPackages_6_1_hardened.xone (linuxKernel.packages.linux_6_1_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_6.xone</li>
    <li>linuxPackages_6_6_hardened.xone (linuxKernel.packages.linux_6_6_hardened.xone)</li>
    <li>linuxPackages_ham.xone (linuxKernel.packages.linux_ham.xone)</li>
    <li>linuxPackages_latest-libre.xone (linuxKernel.packages.linux_latest_libre.xone)</li>
    <li>linuxPackages-libre.xone (linuxKernel.packages.linux_libre.xone)</li>
    <li>linuxPackages_lqx.xone (linuxKernel.packages.linux_lqx.xone)</li>
    <li>linuxPackages_xanmod.xone (linuxKernel.packages.linux_xanmod.xone)</li>
    <li>linuxPackages_xanmod_latest.xone (linuxKernel.packages.linux_xanmod_latest.xone, linuxPackages_xanmod_stable.xone)</li>
    <li>linuxPackages_zen.xone (linuxKernel.packages.linux_zen.xone)</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 16 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.xone</li>
    <li>linuxKernel.packages.linux_5_10_hardened.xone</li>
    <li>linuxKernel.packages.linux_5_4.xone</li>
    <li>linuxKernel.packages.linux_5_4_hardened.xone</li>
    <li>linuxKernel.packages.linux_lqx.xone</li>
    <li>linuxKernel.packages.linux_xanmod.xone</li>
    <li>linuxKernel.packages.linux_xanmod_latest.xone</li>
    <li>linuxKernel.packages.linux_xanmod_stable.xone</li>
    <li>linuxKernel.packages.linux_zen.xone</li>
    <li>linuxPackages_5_10_hardened.xone</li>
    <li>linuxPackages_5_4_hardened.xone</li>
    <li>linuxPackages_lqx.xone</li>
    <li>linuxPackages_xanmod.xone</li>
    <li>linuxPackages_xanmod_latest.xone</li>
    <li>linuxPackages_xanmod_stable.xone</li>
    <li>linuxPackages_zen.xone</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.xone</li>
    <li>linuxPackages_5_15_hardened.xone (linuxKernel.packages.linux_5_15_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_1.xone</li>
    <li>linuxPackages.xone (linuxKernel.packages.linux_6_12.xone)</li>
    <li>linuxPackages_hardened.xone (linuxPackages_6_12_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_13.xone</li>
    <li>linuxPackages_6_13_hardened.xone (linuxKernel.packages.linux_6_13_hardened.xone)</li>
    <li>linuxPackages_latest.xone (linuxKernel.packages.linux_6_14.xone)</li>
    <li>linuxPackages_6_1_hardened.xone (linuxKernel.packages.linux_6_1_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_6.xone</li>
    <li>linuxPackages_6_6_hardened.xone (linuxKernel.packages.linux_6_6_hardened.xone)</li>
    <li>linuxPackages_ham.xone (linuxKernel.packages.linux_ham.xone)</li>
    <li>linuxPackages_latest-libre.xone (linuxKernel.packages.linux_latest_libre.xone)</li>
    <li>linuxPackages-libre.xone (linuxKernel.packages.linux_libre.xone)</li>
  </ul>
</details>
